### DISCUSSION TITLE
NetBeans will not start after upgrade.

### DISCUSSION THREAD

**Discussion by ptoye** [apache/netbeans#4799] [Q&A] [answered]
I have not used NetBeans for some months, when I was using version 12. This was giving some problems, so I decided to upgrade to version 15 so installed it. When opening it for the first time it asked if I wanted to install some plugins from my previous versions and restart NB. Clicked on 'yes' (or whatever it was, I can't remember the details). NBSplash screen appeared and disappeared, but NB did not start.

I uninstalled NB v 15 and tried again. This time, the window about plugins didn't appear, but again NB did not start.

I am now completely stuffed - NB 12 gives problems, NB 15 won't start. I'm attaching an MP4 of my attempt.

https://user-images.githubusercontent.com/16099968/196039451-edccf4ac-d145-4374-8f5b-ac65b8417aa5.mp4

Can someone please help me in getting NB 15 working?

**Comment by mbien** [comment]
did you import the settings from NetBeans 12? 

@Chris2011 do you know where the settings folder is placed on windows? I always forget. Clearing the settings will probably fix this issue.

**Reply by ptoye** [reply]
Yes,  But after uninstalling and reinstalling it didn't ask me again - maybe it had already built a new profile and the uninstall didn't delete it - bad programming.

**Reply by mbien** [reply]
If you import the settings it imports it into the new config folder. Uninstalling won't remove settings as far as I know (I don't use the installer). You are welcome to contribute improvements to the installers or how settings are imported.

**Reply by Chris2011** [reply]
Here it is: `C:\Users\{User}\AppData\Roaming\Netbeans\15\config`

**Reply by Chris2011** [reply]
Just rename the folder to not loose any settings. Just in case.

**Comment by neilcsmith-net** [comment, marked as answer]
This is probably because you imported the nb-javac plugins.  Unfortunately that doesn't work.  Find and delete the user and cache folders (in APPDATA) and try again. For further info on locations see https://github.com/apache/netbeans/blob/master/nb/ide.launcher/netbeans.conf#L21

**Reply by ptoye** [reply]
That's a good idea. Thanks a lot.
I tried it, and renamed the user & cache folders (why the uninstall didn't delete them I don't know). Then installed NB without the java plugins. Now NB is starting OK but of course I've lost all my settings and history, not to mention the CVS plugin (which I can live without).

**Reply by Chris2011** [reply]
I'm thinking about a better "Import settings" dialog where you can do the same as now without changes or with a checkbox, where you can choose what to import + a list of all plugin settings if possible. But out of scope here.

**Reply by ptoye** [reply]
I'd live to contribute in a practical sense to the project. However, my health is not good and there's no point in starting something that I won't be able to finish. It would probably take me a month or two to find out how the installer works, and I don't think  I have that timescale.

It worries me that new releases seem to keep breaking things. I've another thread where previously working code stopped working (that's not in NetBeans but Visual Studio). Depressing.
